### PR TITLE
Add detailed Python support documentation

### PR DESCRIPTION
**Pull request by milanwilliams**
# Thanks for improving Semgrep Docs 😀

(wip, need to add the nav bar but would appreciate an early review 🙇 )

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: docs/supported-languages.md [hunk 1]
```diff
@@ -103,8 +106,6 @@ Community supported languages have varying levels of rule coverage - view the re
  <summary>Click to view <strong>Semgrep OSS</strong> languages.</summary>
 
 - Bash
-- C
```
Comment (s-santillan) [inline docs/supported-languages.md:106]: @milanwilliams Just to confirm, are we absolutely removing all language parsing capabilities for C & C++? Put another way, are we **gating** scans for C & C++, i.e the user has to login?

**Comment by s-santillan** [comment]
Hi @milanwilliams This is looking really good! I love the benchmark information and details, exactly the sort of thing people want to see when evaluating our performance for a language.

(1) 
To view this page live, I added it in `sidebars.js`, I hope that's okay. I didn't make any other edits without your consent. 

(2) 
May I suggest that in the future we add information for Supply Chain here as well? I know our competitor(s) piece out the documentation into several docs per product, but I'm under the impression we want a unified Semgrep product and it's 
 nice for the user to see all supported products for 1 language in 1 place. WDYT?

File: docs/semgrep-code/supported-languages-python.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+| 109 | xml.sax                  | XML Parsing Library                      |
+
+
+</details>
+
+### Benchmark results (_exclusive of_ [AI](https://semgrep.dev/docs/semgrep-assistant/overview) processing): 
+
+Our benchmarking process involves scanning open-source repositories, triaging the findings, and making iterative rule updates. This process was developed and is used internally by our security research team to monitor and improve rule performance.
+
+Results as of **2024-09-04**:
```
Comment (s-santillan) [inline docs/semgrep-code/supported-languages-python.md:173]:
```suggestion
Results as of **September 9, 2024**:
```
It is recommended by the google style guide to spell out [the date in full](https://developers.google.com/style/dates-times#why-dates-written-out) if we can:
image

File: docs/semgrep-code/supported-languages-python.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    </tr></thead>
+    <tbody>
+    <tr>
+        <td>Framework / library</td>
+        <td>Django</td>
+        <td>Flask</td>
+        <td>FastAPI</td>
+    </tr>
+    <tr>
+        <td>Category</td>
```
Comment (s-santillan) [inline docs/semgrep-code/supported-languages-python.md:42]:
```suggestion
        <td colspan="3">Category</td>
```
May resolve the following blank cells
image

File: docs/semgrep-code/supported-languages-python.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+Semgrep’s Python coverage leverages framework-specific analysis capabilities that are not present in OSS. As a result, many framework specific Pro rules will **fail** to return findings if run on OSS. To ensure full security coverage, run: `semgrep login && semgrep ci`
+:::
+
+## Analysis Capabilities for Python
+* Inter-file analysis (cross-file)
+* Inter-procedural analysis (cross-function)
+* Framework-specific control flow analysis 
+
+## Coverage 
+_Our coverage ethos: comprehensive and accurate detection of common, OWASP Top 10 issues in source code_
```
Comment (s-santillan) [inline docs/semgrep-code/supported-languages-python.md:24]:
```suggestion
_Semgrep's coverage ethos: comprehensive and accurate detection of common, OWASP Top 10 issues in source code_
```
image

File: docs/semgrep-code/supported-languages-python.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  Detailed documentation for Semgrep's Python support. 
+hide_title: true
+tags:
+    - Semgrep Code 
+title: Supported languages for Python
+---
+# Python
+
+:::tip 
+Semgrep’s Python coverage leverages framework-specific analysis capabilities that are not present in OSS. As a result, many framework specific Pro rules will **fail** to return findings if run on OSS. To ensure full security coverage, run: `semgrep login && semgrep ci`
```
Comment (s-santillan) [inline docs/semgrep-code/supported-languages-python.md:15]:
```suggestion
Semgrep’s Python coverage leverages framework-specific analysis capabilities that are not present in OSS. As a result, many framework specific Pro rules will **fail** to return findings if run on OSS. To ensure full security coverage, run: `semgrep login && semgrep ci`.
```

File: docs/semgrep-code/supported-languages-python.md [hunk 6]
```diff
@@ -16,15 +16,15 @@ Semgrep’s Python coverage leverages framework-specific analysis capabilities t
...
 * Inter-file analysis (cross-file)
 * Inter-procedural analysis (cross-function)
-* Framework-specific control flow analysis 
 
 ## Coverage 
 _Our coverage ethos: comprehensive and accurate detection of common, OWASP Top 10 issues in source code_
 
 
-Semgrep supports **over 100** common Python frameworks and libraries, including the 3 most popular frameworks: 
+Semgrep Code has native support for the most common web frameworks:
```
Comment (s-santillan) [inline docs/semgrep-code/supported-languages-python.md:27]: @chushili1 What does "native support" mean?

Comment (chushili1) [reply docs/semgrep-code/supported-languages-python.md:27]: Native support means that in addition to rules, the engine itself has built out capabilities for handling implicit dataflows in these frameworks (Django, Flask, FastAPI)

Comment (s-santillan) [reply docs/semgrep-code/supported-languages-python.md:27]: Note for me to help with the rephrasing: 
image

Comment (s-santillan) [reply docs/semgrep-code/supported-languages-python.md:27]:
```suggestion
In addition to rules, the Semgrep engine itself can analyze code and implicit dataflows in the context of the following supported frameworks:
```
Or something similar!

**Comment by milanwilliams** [comment]
> Hi @milanwilliams This is looking really good! I love the benchmark information and details, exactly the sort of thing people want to see when evaluating our performance for a language.
> 
> (1) To view this page live, I added it in `sidebars.js`, I hope that's okay. I didn't make any other edits without your consent.
>  Thank you!!
> (2) May I suggest that in the future we add information for Supply Chain here as well? I know our competitor(s) piece out the documentation into several docs per product, but I'm under the impression we want a unified Semgrep product and it's nice for the user to see all supported products for 1 language in 1 place. WDYT?
Absolutely. If folks like it, I'd love to do this for all languages + x-product!

**Review by s-santillan — APPROVED** [review]

**Comment by s-santillan** [comment]
In case anyone wants to see the preview, btw: https://deploy-preview-1735--semgrep-docs-prod.netlify.app/docs/semgrep-code/supported-languages-python

**Pull request merged**